### PR TITLE
docs: norwegian translation

### PR DESCRIPTION
**Pull request by estubmo**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Norwegian translation

* [x]  deployment
    * [x]  docker.md
    * [x]  vercel.md
* [x]  usage
  * [x]  env-variables.md
  * [x]  first-steps.md
  * [x]  next-auth.md
  * [x]  next-js.md
  * [x]  prisma.md
  * [x]  tailwind.md
  * [x]  trpc.md
  * [x]  typescript.md
 * [x]  faq.md
 * [x]  folder-structure.md
 * [x]  installation.md
 * [x]  introduction.md
 * [x]  other-recs.md
 * [x]  t3-collection.md
 * [x]  why.md

---

## Screenshots

image

### CODE REVIEW

File: www/src/pages/en/usage/trpc.md [hunk 1]
```diff
@@ -230,7 +230,7 @@ Compare this to the tRPC example above and you can see some of the advantages of
 
 Here are some snippets that might come in handy.
 
-### Enabling CORS
+### Aktivering av CORS
 
 If you need to consume your API from a different domain, for example in a monorepo that includes a React Native app, you might need to enable CORS:
```
Comment (josephayman) [inline www/src/pages/en/usage/trpc.md:235]:
```suggestion
Hvis du trenger å bruke API-en din fra et annet domene, for eksempel i en monorepo som inkluderer en React Native-app, må du kanskje aktivere CORS:
```

File: www/src/pages/no/deployment/docker.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
+---
+title: Docker
+description: Utrulling med Docker
+layout: ../../../layouts/docs.astro
+lang: no
+---
+
+Stakken kan rulles ut med Docker. Enten som en enkel beholder eller som en gruppe beholdere ved hjelp av `docker-compose`. Se [`ajcwebdev/ct3a-docker`](https://github.com/ajcwebdev/ct3a-docker) for et eksempel-_repo_ som er basert på denne dokumentasjonen.
```
Comment (josephayman) [inline www/src/pages/no/deployment/docker.md:8]:
```suggestion
Stakken kan rulles ut med Docker. Enten som en enkel kontainer eller som en gruppe kontainere ved hjelp av `docker-compose`. Se [`ajcwebdev/ct3a-docker`](https://github.com/ajcwebdev/ct3a-docker) for et eksempel-_repo_ som er basert på denne dokumentasjonen.
```

File: www/src/pages/no/deployment/docker.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+.git
+```
+
+</div>
+
+</details>
+
+### 3. Lag Dockerfile
+
+> Siden vi ikke drar serverens miljøvariabler inn i beholderen, vil [skjema for validering av miljøvariabler](/no/usage/env-variables) feile. For å forhindre dette må vi legge til `SKIP_ENV_VALIDATION=1` i byggkommandoen slik at miljøvariabelskjemaene ikke valideres ved "build time".
```
Comment (josephayman) [inline www/src/pages/no/deployment/docker.md:54]:
```suggestion
> Siden vi ikke drar serverens miljøvariabler inn i kontaineren, vil [skjema for validering av miljøvariabler](/no/usage/env-variables) feile. For å forhindre dette må vi legge til `SKIP_ENV_VALIDATION=1` i byggkommandoen slik at miljøvariabelskjemaene ikke valideres ved "build time".
```

File: www/src/pages/no/deployment/docker.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+```bash
+docker build -t ct3a-docker --build-arg NEXT_PUBLIC_CLIENTVAR=clientvar .
+docker run -p 3000:3000 -e DATABASE_URL="database_url_her" ct3a-docker
+```
+
+Åpne [localhost:3000](http://localhost:3000/) for å se programmet som kjøres.
+
+## Docker Compose
+
+Du kan også bruke Docker Compose for å bygge bildet og kjøre det i beholderen.
```
Comment (josephayman) [inline www/src/pages/no/deployment/docker.md:151]:
```suggestion
Du kan også bruke Docker Compose for å bygge bildet og kjøre det i kontaineren.
```

File: www/src/pages/en/usage/trpc.md [hunk 5]
```diff
@@ -230,7 +230,7 @@ Compare this to the tRPC example above and you can see some of the advantages of
 
 Here are some snippets that might come in handy.
```
Comment (josephayman) [inline www/src/pages/en/usage/trpc.md:231]:
```suggestion
Her er noen kodesnutter som kan hjelpe deg.
```

Comment (josephayman) [reply www/src/pages/en/usage/trpc.md:231]: Is this file a part of the translation? since it's under en/usage

Comment (estubmo) [reply www/src/pages/en/usage/trpc.md:231]: No, that's an error.

File: www/src/pages/en/usage/trpc.md (hunk 1 above)
Comment (josephayman) [reply www/src/pages/en/usage/trpc.md:235]: You are right.

File: www/src/pages/no/usage/next-auth.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+Det samme mønsteret kan brukes til å legge til flere data til `session`-objektet, som f.eks et `role`-felt, men **skal ikke misbrukes til å lagre sensitive data på klienten**.
+
+## Bruk med tRPC
+
+Hvis du bruker NextAuth.js med tRPC, kan du opprette gjenbrukbare beskyttede prosedyrer med [middlewares](https://trpc.io/docs/v10/middlewares). Dette lar deg lage prosedyrer som bare er tilgjengelige for autentiserte brukere. `create-t3-app` gir deg allerede dette, slik at du enkelt kan få tilgang til session-objektet i autentiserte prosedyrer.
+
+Dette skjer i to trinn:
+
+1. Få tilgang til sesjonen fra _request-headerne_ ved å bruke funksjonen [`unstable_getServerSession`](https://next-auth.js.org/configuration/nextjs#unstable_getserversession). Ikke bekymre deg, denne funksjonen er trygg å bruke - navnet inneholder bare `ustabil` fordi API-implementeringen kan endres i fremtiden. Fordelen med `unstable_getServerSession` sammenlignet med `getSession` er at det er en funksjon på serversiden og medfører ikke unødvendige kall. `create-t3-app` lager en hjelpefunksjon som abstraherer dette særegne API-et.
```
Comment (josephayman) [inline www/src/pages/no/usage/next-auth.md:74]:
```suggestion
1. Få tilgang til sesjonen fra _request-headerne_ ved å bruke funksjonen [`unstable_getServerSession`](https://next-auth.js.org/configuration/nextjs#unstable_getserversession). Ikke bekymre deg, denne funksjonen er trygg å bruke - navnet inneholder bare `unstable` fordi API-implementeringen kan endres i fremtiden. Fordelen med `unstable_getServerSession` sammenlignet med `getSession` er at det er en funksjon på serversiden og medfører ikke unødvendige kall. `create-t3-app` lager en hjelpefunksjon som abstraherer dette særegne API-et.
```

File: www/src/pages/no/usage/next-auth.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+2. Lag en tRPC-middleware som sjekker om brukeren er autentisert. Vi bruker deretter middlewaren i en `protectedProcedure`. Hvert kall av disse prosedyrene må autentiseres, ellers kastes en feilmelding, som kan håndteres av klienten.
+
+```ts:server/trpc/trpc.ts
+const isAuthed = t.middleware(({ ctx, next }) => {
+  if (!ctx.session || !ctx.session.user) {
+    throw new TRPCError({ code: "UNAUTHORIZED" });
+  }
+  return next({
+    ctx: {
+      // inferer `session` somm ikke-nullbar
```
Comment (josephayman) [inline www/src/pages/no/usage/next-auth.md:108]:
```suggestion
      // inferer `session` som ikke-nullbar
```

**Review by josephayman — APPROVED** [review]

**Comment by juliusmarminge** [comment]
Can you implement the new folder-structure page? https://create.t3.gg/en/folder-structure

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**